### DISCUSSION TITLE
Cannot kill a machine

### DISCUSSION THREAD

**Discussion by kchen1023** [cmu-sei/GHOSTS#366] [Q&A] [answered]
Hi, I'm currently writing python scripts that uses the API. I am able to make machines, but I cannot kill them or send them timelines.
1. When I request to kill the machine, it returns 204 instead of 200.
2. When I request to send a timeline to a machine, it returns with a 204 instead of 200.
3. When I request to get all timelines, it returns an empty list.
Are these the intended behaviors? (Sorry if I said something dumb, I'm still quite new...)

**Comment by kchen1023** [comment]
Also, what is the range that this is refering to?
![image](https://github.com/cmu-sei/GHOSTS/assets/131231802/42fd71a0-f3a7-4113-af3c-223684fbaf7d)

**Reply by sei-dupdyke** [reply, marked as answer]
"on the range" means the client machines you have setup to check-in with the ghosts api. Typically this is done in a virtual machine "range".

**Reply by kchen1023** [reply]
I'm currently working on a Linux. When I set up a client (and only client) on Linux and try to run (with 'dotnet ghosts.client.linux.dll'), it doesn't work. I cannot even connect to localhost:5000.
But after installing the API server on the same system, it worked. 
I thought the server was not necessary to run the client?

**Reply by sei-dupdyke** [reply]
It's not — the client will still continue to perform its activity, it just won't report in, nor can it be controlled via the API, but otherwise, it'll do the things you configure it to do!
